### PR TITLE
 firecracker: Add support for default VM configuration

### PR DESCRIPTION
**Pull request by mcastelino**
firecracker: Add support for default VM configuration

Kata support specifing the default VM configuration via configuration.toml. This allows the system or cluster admin  to choose the default (i.e minimum) size of the VM.

Add support in kata to respect the VM configuration for firecracker.

Fixes: #1594

Signed-off-by: Manohar Castelino <manohar.r.castelino@intel.com>

### CODE REVIEW

File: virtcontainers/fc.go [hunk 1]
```diff
@@ -351,6 +373,10 @@ func (fc *firecracker) startSandbox(timeout int) error {
 		return err
 	}
 
+	fc.fcSetVMBaseConfig(int64(fc.config.MemorySize),
```
Comment (sboeuf) [inline virtcontainers/fc.go]: Please check the error returned here.

File: virtcontainers/fc.go [hunk 2]
```diff
@@ -315,6 +315,28 @@ func (fc *firecracker) fcSetVMRootfs(path string) error {
...
+		MemSizeMib: mem,
+		VcpuCount:  vcpus,
+	}
+	param.SetBody(cfg)
+	_, err := fc.client().Operations.PutMachineConfiguration(param)
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (sboeuf) [inline virtcontainers/fc.go]: You can simplify this:
```go
        _, err := fc.client().Operations.PutMachineConfiguration(param)
	if err != nil {
		return err
	}

	return nil
```
with this:
```go
        _, err := fc.client().Operations.PutMachineConfiguration(param)
        return err	
```

**Comment by grahamwhaley** [comment]
holding off a \ t e s t and merge until we hear back from @mcastelino

File: virtcontainers/fc.go (hunk 1 above)
Comment (mcastelino) [reply virtcontainers/fc.go]: done

File: virtcontainers/fc.go (hunk 2 above)
Comment (mcastelino) [reply virtcontainers/fc.go]: Done

**Comment by jcvenegas** [comment]
Fedora job failed due to: https://github.com/kata-containers/tests/issues/1449
```
• Failure [22.547 seconds]
state
/tmp/jenkins/workspace/kata-containers-runtime-fedora-PR/go/src/github.com/kata-containers/tests/functional/state_test.go:26
  container
  /tmp/jenkins/workspace/kata-containers-runtime-fedora-PR/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
    with workload [true], timeWait 5 [It]
    /tmp/jenkins/workspace/kata-containers-runtime-fedora-PR/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46

    Expected
        <int>: 1
    to equal
        <int>: 0

    /tmp/jenkins/workspace/kata-containers-runtime-fedora-PR/go/src/github.com/kata-containers/tests/functional/state_test.go:45
------------------------------
SSSSSSS

Summarizing 1 Failure:

[Fail] state container [It] with workload [true], timeWait 5 
/tmp/jenkins/workspace/kata-containers-runtime-fedora-PR/go/src/github.com/kata-containers/tests/functional/state_test.go:45

Ran 1 of 8 Specs in 22.548 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 7 Skipped --- FAIL: TestFunctional (22.55s)
```

**Pull request merged**